packages/server/src/api/students.rs

```diff
@@ -137,8 +137,22 @@ pub async fn compute_degree_status(mut user: User, db: Data<Db>) -> Result<HttpR
 
     user.details.modified = false;
 
-    let vec_courses = db
-        .get_filtered::<Course>(catalog.get_all_course_ids(), FilterType::In, "_id")
+    let courses = db
+        .get_filtered::<Course>(
+            FilterType::In,
+            "_id",
+            catalog
+                .get_all_course_ids()
+                .into_iter()
+                .chain(
+                    user.details
+                        .degree_status
+                        .course_statuses
+                        .iter()
+                        .map(|cs| cs.course.id.clone()),
+                )
+                .collect::<Vec<CourseId>>(),
+        )
```

Comment:
It can create duplications in `courses`, can't it?
I think we should consider to collect it to a `Set` and then change it again to vector.